### PR TITLE
Language support

### PR DESCRIPTION
**Pull request by MikkoLipsanen**
Added language support (Finnish, Swedish, English) for buttons and stable text elements in frontend components
Added tests for the language support functionality
Highlighted the feedback text element

### CODE REVIEW

File: src/fs-result.js [hunk 1]
```diff
@@ -151,6 +152,32 @@ class FsResult extends HTMLElement {
...
+            this.language = newValue;
+            this.setLanguage();
+        }
+    }
+
+    setLanguage() {
+        if (this.language === 'en') {
+            this.headerText = 'Results';
+        } else if (this.language === 'sv') {
+            this.headerText = 'Resultater';
```
Comment (EssKayz) [inline src/fs-result.js]:
```suggestion
            this.headerText = 'Resultat';
```

File: test/fs-result.test.js [hunk 2]
```diff
@@ -102,3 +102,41 @@ describe('List element test suite', () => {
...
+
+            expect(el.shadowRoot.querySelector('.blue').textContent).to.contain('Results');
+        });
+
+        it('to Swedish', async () => {
+            const el = await fixture('<fs-result language="sv"></fs-result>');
+
+            await sleep(100);
+
+            expect(el.shadowRoot.querySelector('.blue').textContent).to.contain('Resultater');
```
Comment (EssKayz) [inline test/fs-result.test.js]:
```suggestion
            expect(el.shadowRoot.querySelector('.blue').textContent).to.contain('Resultat');
```

File: src/fs-question.js [hunk 3]
```diff
@@ -350,6 +375,11 @@ class FsQuestion extends HTMLElement {
     }
 }
 
+const languages = {
+    'fi': {'yesText': 'Kyllä', 'noText': 'Ei', 'skipText': 'Ohita', 'previousText': 'Edellinen', 'nextText': 'Seuraava'},
+    'en': {'yesText': 'Yes', 'noText': 'No', 'skipText': 'Skip', 'previousText': 'Previous', 'nextText': 'Next'},
+    'sv': {'yesText': 'Ja', 'noText': 'Nej', 'skipText': 'Håppa över', 'previousText': 'Föregående', 'nextText': 'Nästa'}};
```
Comment (EssKayz) [inline src/fs-question.js]:
```suggestion
    'sv': {'yesText': 'Ja', 'noText': 'Nej', 'skipText': 'Hoppa över', 'previousText': 'Föregående', 'nextText': 'Nästa'}};
```

File: test/fs-question.test.js [hunk 4]
```diff
@@ -316,3 +316,112 @@ describe('Multiquestion POST:s', async () => {
...
+
+            await sleep(100);
+
+            const okButton = element.shadowRoot.querySelector('.ok');
+            okButton.click();
+            await sleep(200);
+
+            expect(okButton.textContent).to.eq('Ja');
+            expect(element.shadowRoot.querySelector('.no').textContent).to.eq('Nej');
+            expect(element.shadowRoot.querySelector('.skip').textContent).to.eq('Håppa över');
```
Comment (EssKayz) [inline test/fs-question.test.js]:
```suggestion
            expect(element.shadowRoot.querySelector('.skip').textContent).to.eq('Hoppa över');
```

File: test/fs-question.test.js [hunk 5]
```diff
@@ -316,3 +316,112 @@ describe('Multiquestion POST:s', async () => {
...
+            nextButton.click();
+            await sleep(200);
+
+            expect(nextButton.textContent).to.eq('Nästa');
+            expect(element.shadowRoot.querySelector('.previous').textContent).to.eq('Föregående');
+            const table = element.shadowRoot.querySelector('.multi-table');
+
+            expect(table).to.contain.html('Ja');
+            expect(table).to.contain.html('Nej');
+            expect(table).to.contain.html('Håppa över');
```
Comment (EssKayz) [inline test/fs-question.test.js]:
```suggestion
            expect(table).to.contain.html('Hoppa över');
```

File: src/fs-detail.js [hunk 6]
```diff
@@ -217,4 +252,17 @@ class FsDetail extends HTMLElement {
...
     }
 }
+
+const languages = {
+    'fi': {'yesButton': 'Kyllä', 'noButton': 'Ei', 'feedbackQuestion': 'Oliko tämä hakemanne luokka?', 'feedbackReply': 'Kiitos palautteestanne!',
+        'excludesText': 'Tähän ei kuulu: ', 'includesText': 'Tähän kuuluu: ', 'includesAlsoText': 'Tähän kuuluu myös: ', 'keywordsText': 'Synonyymit: '},
+    'en': {'yesButton': 'Yes', 'noButton': 'No', 'feedbackQuestion': 'Is this the building class you are looking for?',
+        'feedbackReply': 'Thank you for your feedback!', 'excludesText': 'Excludes: ', 'includesText': 'Includes: ', 'includesAlsoText': 'Includes also:',
+        'keywordsText': 'Keywords: '},
+    'sv': {'yesButton': 'Ja', 'noButton': 'Nej', 'feedbackQuestion': 'Är detta byggnadsklassen ni var efter?', 'feedbackReply':
```
Comment (xTooth) [inline src/fs-detail.js]:
```suggestion
    'sv': {'yesButton': 'Ja', 'noButton': 'Nej', 'feedbackQuestion': 'Är detta byggnadsklassen ni var ute efter?', 'feedbackReply':
```
simple rewording

File: src/fs-detail.js [hunk 7]
```diff
@@ -217,4 +252,17 @@ class FsDetail extends HTMLElement {
...
+        'Tack för responsen!', 'excludesText': 'Innehåller inte: ', 'includesText': 'Innehåller: ', 'includesAlsoText': 'Innehåller också: ',
```
Comment (xTooth) [inline src/fs-detail.js]:
```suggestion
        'Tack för responsen!', 'excludesText': 'Innehåller ej: ', 'includesText': 'Innehåller: ', 'includesAlsoText': 'Innehåller också: ',
```
again just a simple rewording

File: test/fs-detail.test.js [hunk 8]
```diff
@@ -119,3 +119,61 @@ describe('Feedback test run', () => {
...
+            await sleep(100);
+
+            element.updateDetail(complexClassification);
+            const sr = element.shadowRoot;
+
+            const okButton = sr.querySelector('.ok');
+
+            expect(okButton.textContent).to.eq('Ja');
+            expect(sr.querySelector('.no').textContent).to.eq('Nej');
+            expect(sr.querySelector('.feedback')).to.contain.html('Är detta byggnadsklassen ni var efter?');
```
Comment (xTooth) [inline test/fs-detail.test.js]:
```suggestion
            expect(sr.querySelector('.feedback')).to.contain.html('Är detta byggnadsklassen ni var ute efter?');
```

File: test/fs-detail.test.js [hunk 9]
```diff
@@ -119,3 +119,61 @@ describe('Feedback test run', () => {
...
+            expect(sr.querySelector('.white')).to.contain.html('Innehåller inte');
```
Comment (xTooth) [inline test/fs-detail.test.js]:
```suggestion
            expect(sr.querySelector('.white')).to.contain.html('Innehåller ej');
```

File: src/fs-detail.js [hunk 10]
```diff
@@ -217,4 +252,17 @@ class FsDetail extends HTMLElement {
         }
     }
 }
+
+const languages = {
+    'fi': {'yesButton': 'Kyllä', 'noButton': 'Ei', 'feedbackQuestion': 'Oliko tämä hakemanne luokka?', 'feedbackReply': 'Kiitos palautteestanne!',
+        'excludesText': 'Tähän ei kuulu: ', 'includesText': 'Tähän kuuluu: ', 'includesAlsoText': 'Tähän kuuluu myös: ', 'keywordsText': 'Synonyymit: '},
+    'en': {'yesButton': 'Yes', 'noButton': 'No', 'feedbackQuestion': 'Is this the building class you are looking for?',
+        'feedbackReply': 'Thank you for your feedback!', 'excludesText': 'Excludes: ', 'includesText': 'Includes: ', 'includesAlsoText': 'Includes also:',
```
Comment (xTooth) [inline src/fs-detail.js]:
```suggestion
        'feedbackReply': 'Thank you for your feedback!', 'excludesText': 'Excludes: ', 'includesText': 'Includes: ', 'includesAlsoText': 'Also includes:',
```

File: test/fs-detail.test.js [hunk 11]
```diff
@@ -119,3 +119,61 @@ describe('Feedback test run', () => {
...
+            const sr = element.shadowRoot;
+
+            const okButton = sr.querySelector('.ok');
+
+            expect(okButton.textContent).to.eq('Yes');
+            expect(sr.querySelector('.no').textContent).to.eq('No');
+            expect(sr.querySelector('.feedback')).to.contain.html('Is this the building class you are looking for?');
+            expect(sr.querySelector('.white')).to.contain.html('Excludes');
+            expect(sr.querySelector('.white')).to.contain.html('Includes');
+            expect(sr.querySelector('.white')).to.contain.html('Includes also');
```
Comment (xTooth) [inline test/fs-detail.test.js]:
```suggestion
            expect(sr.querySelector('.white')).to.contain.html('Also includes');
```

**Pull request merged**